packages/jamf_compliance_reporter/data_stream/app_metrics/elasticsearch/ingest_pipeline/default.yml

```
@@ -0,0 +1,150 @@
+      ignore_failure: true
+  - convert:
+      field: json.app_metric_info.cpu_percentage
+      type: double
+      target_field: jamf_compliance_reporter.app_metrics.app_metric_info.cpu_percentage
+      ignore_failure: true 
+  - convert:
+      field: json.app_metric_info.cpu_time_seconds
+      type: double
+      target_field: jamf_compliance_reporter.app_metrics.app_metric_info.cpu_time_seconds
+      ignore_failure: true
+  - convert:
+      field: json.app_metric_info.interrupt_wakeups
+      type: long
+      target_field: jamf_compliance_reporter.app_metrics.app_metric_info.interrupt_wakeups
+      ignore_failure: true
+  - convert:
+      field: json.app_metric_info.platform_idle_wakeups
+      type: long
+      target_field: jamf_compliance_reporter.app_metrics.app_metric_info.platform_idle_wakeups
+      ignore_failure: true
+  - convert:
+      field: json.app_metric_info.resident_memory_size_mb
+      type: double
+      target_field: jamf_compliance_reporter.app_metrics.app_metric_info.resident_memory_size_mb
+      ignore_failure: true
+  - convert:
+      field: json.app_metric_info.virtual_memory_size_mb
+      type: double
+      target_field: jamf_compliance_reporter.app_metrics.app_metric_info.virtual_memory_size_mb
+      ignore_failure: true
+  - rename:
+      field: json.header.event_name
+      target_field: jamf_compliance_reporter.header.event_name
+      ignore_missing: true
+  - set:
+      field: event.action
+      copy_from: jamf_compliance_reporter.header.event_name
+      ignore_failure: true
+  - lowercase:
+      field: event.action
+      ignore_missing: true
+  - date:
+      field: json.header.time_seconds_epoch
+      target_field: '@timestamp'
+      if: ctx?.json?.header?.time_seconds_epoch != '0'
+      ignore_failure: true
+      formats:
+        - UNIX
+  - rename:
+      field: json.host_info.host_name
+      target_field: host.hostname
+      ignore_missing: true
+  - append:
+      field: related.hosts
+      value: '{{{host.hostname}}}'
+      if: ctx?.host?.hostname != null
+      allow_duplicates: false
+      ignore_failure: true
+  - rename:
+      field: json.host_info.host_uuid
+      target_field: host.id
+      ignore_missing: true
+  - rename:
+      field: json.host_info.osversion
+      target_field: host.os.version
+      ignore_missing: true
+  - append:
+      field: host.mac
+      value: '{{{json.host_info.primary_mac_address}}}'
+      if: ctx?.json?.host_info?.primary_mac_address != null
+      allow_duplicates: false
+      ignore_failure: true
+  - gsub:
+      field: host.mac
+      pattern: '[-:.]'
+      replacement: '-'
+      ignore_missing: true
+  - uppercase:
+      field: host.mac
+      ignore_missing: true
+  - rename:
+      field: json.host_info.serial_number
+      target_field: jamf_compliance_reporter.host_info.serial_number
+      ignore_missing: true
+  - script:
+      lang: painless
+      if: ctx?.json?.app_metric_info?.cpu_percentage != null
```

```suggestion
      if: ctx.json?.app_metric_info?.cpu_percentage != null
```

`ctx` is always going to be present. Can you apply this change everywhere.